core/che-core-metrics-core/src/main/java/org/eclipse/che/core/metrics/MetricsServer.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.core.metrics;
+
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.exporter.HTTPServer;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Bind on 8087 port http endpoint with prometheus metrics. */
+@Singleton
+public class MetricsServer {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MetricsServer.class);
+
+  private HTTPServer server;
+
+  @Inject
+  public void startServer(CollectorRegistry collectorRegistry) throws IOException {
+    this.server = new HTTPServer(new InetSocketAddress(8087), collectorRegistry, true);
```

Comment:
Shouldn't the port be configurable?

Reply:
In theory, it can. Is there any practical reason to do that from your pov? We discussed that with @sleshchenko and thought to add, but didn't find the reason what for, and for example why 8080 not configured.

Reply:
I was just coming to it from the side of "if devfile schema location is configurable, then this should be, too". The only practical reason would be if it was deployed co-located in the same pod as other service wanting the same port.